### ISSUE TITLE
[Feedback] Virtual cards 

### ISSUE THREAD

**Issue by BenJam** [opencollective/opencollective#4312]
> Access funds managed through Open Collective with Virtual Cards

![Screenshot 2021-05-17 at 17 07 25](https://user-images.githubusercontent.com/158833/119471264-0804fa80-bd41-11eb-892b-51681d32c365.png)

We're lowering the barriers to access the financial system even further, with Virtual Card integration from Privacy.com. To get started click 'request a card' in the actions menu on your Collective's profile page. 

**[Read more about virtual cards](https://blog.opencollective.com/virtual-card-support/)**
**Check out the documentation [for collectives](https://docs.opencollective.com/help/expenses-and-getting-paid/virtual-cards) and [for hosts](https://docs.opencollective.com/help/fiscal-hosts/virtual-cards)**

**Comment by BenJam** [comment]
I mainly created this issue to gather some ideas for improvements myself, but I noticed Alanna provided some in https://github.com/opencollective/opencollective/issues/4261

**Comment by BenJam** [comment]
I thought I'd lef ta big ol list of feedback about this previoudly but I guess I lost it in an open tab :facepalm: 

- include assignee, monthly limit and name in request
- make a request for a non-vendor-locked card (because collectives are making ad hoc one off payments)
- combine the collective filter on the host dashboard with the others (so i dont have to click)
- include the billing details in collective's virtual card page
- allow collectives to permanantly cancel cards (so the host can use the quoate again)
- include a quick link to 'assign card' for the collective in the request email
- icnlude a quick view of cards already assigned to a collective when about to assign a card (because I need to know whether someone else already assigned a card)

**Comment by BenJam** [comment]
one teething problem that we have with the process: privacy is currnetly blocking card creation because we have a number of cards that are still unused. So we may need to look at providing cards at another entry point (maybe where we spot recurring expenses) and/or work with Privacy/another provider on provisioning policy.

**Comment by BenJam** [comment]
So after a few days of ☝️ I think i need a place for collectives that have requested a card so I can go back to them and assign quickly.

**Comment by BenJam** [comment]
If we stay with Privacy it's likely that we should add a 'unused' indicator to cards so that hosts can get in touch and cancel cards to free up their quota. I do not recommend adding a feature to re-assign on the off chance that we later sync a payment that has been made.

**Comment by BenJam** [comment]
Add to the list: collectives that have not been accepted should not be able to request virtual cards.

**Comment by BenJam** [comment]
Looks like we might have a problem with google's domain registrar and virtual cards https://opencollective.freshdesk.com/a/tickets/21613

**Comment by alanna** [comment]
Fiscal hosts should be checking and approving card receipts. Right now people could upload any file as an attachment and we'd never catch it.

**Comment by BenJam** [comment]
I'm going to start winding this up into something of a 'next steps' but we had one great bit of feedback in slack:

> As an admin it would make my life immensely easier if you could have some kind of indicator on the expenses layout to show which ones need to have a receipt uploaded.

I dare say that this is generally a good thing to have for _any_ expense, rather than just virtual cards. In case we add any features in the future that allow folks to open an expense and add a receipt later.

**Comment by BenJam** [comment]
okay just to wind this up, I think before we push this out to more collectives we need to:

- stop collectives that have not been accepted requesting a card
- stop collectives without a balance requesting a card
- show the number of cards that are already allocated when allocating a new card
- add messaging re. billing details, 
- pausing all cards once the balance goes to zero

nice to haves would include

- adding a card limit to card requests
- amending budgets on cards allocated as balance approaches the amount on limits for all cards issued

**Comment by kaylarep** [comment]
![image](https://user-images.githubusercontent.com/64920686/126707788-d7972b61-c92e-41e3-94f2-8009a7ffa8b5.png)

Would love some 'Sort by' options: Date created, alphabetical

The 'Specific Collectives' List should probably be alphabetical if possible

Also View __ #of cards on page would be helpful.

**Comment by kaylarep** [comment]
https://github.com/opencollective/opencollective/issues/4556

**Comment by kaylarep** [comment]
ended up on this page https://opencollective.com/foundation/edit/virtual-cards through an email notification, wondering how i could have accessed it thru the settings menu. (the host's own virtual cards that are assigned to themselves (not one of their hosted groups, so Settings-> Virtual Cards wont take me here)

**Comment by kaylarep** [comment]
some typos on https://opencollective.com/foundation/edit/virtual-cards

**Comment by maximiliano1993** [comment]
> > Accede a los fondos gestionados a través de Open Collective con Tarjetas Virtuales
> 
> ![Captura de pantalla del 17 de mayo de 2021 a las 17 07 25](https://user-images.githubusercontent.com/158833/119471264-0804fa80-bd41-11eb-892b-51681d32c365.png)
> 
> Estamos reduciendo aún más las barreras para acceder al sistema financiero, con la integración de Tarjeta Virtual de Privacy.com. Para comenzar, haga clic en 'solicitar una tarjeta' en el menú de acciones en la página de perfil de su Colectivo.
> 
> **[Leer más sobre tarjetas virtuales](https://blog.opencollective.com/virtual-card-support/)** ** Consulte la documentación [para colectivos](https://docs.opencollective.com/help/expenses-and-getting-paid/virtual-cards) y [para hosts](https://docs.opencollective.com/help/fiscal-hosts/virtual-cards)**

max

**Comment by kaylarep** [comment]
not sure if this is happening just on my computer, but when I have specific collectives selected to view, it cuts off their names:
![image](https://user-images.githubusercontent.com/64920686/140235854-3f34bab3-c399-415f-8cb1-89aa14db6dd7.png)

**Comment by Betree** [comment]
@SudharakaP Could [this](https://github.com/opencollective/opencollective/issues/4312#issuecomment-960309350) be related to the update on StyledSelect? Like we only show the images, but somehow they're not loaded so we're showing the default avatars?

**Comment by SudharakaP** [comment]
> @SudharakaP Could [this](https://github.com/opencollective/opencollective/issues/4312#issuecomment-960309350) be related to the update on StyledSelect? Like we only show the images, but somehow they're not loaded so we're showing the default avatars?

Yes, this is related to that for sure. The compact mode only shows avatars if the avatars are available and loaded. I didn't handle this case as I thought it's acceptable since the user knows what they are choosing (using the dropdown you see collective) and then if they hover over the selected ones, it  shows the full name. 🤔

**Comment by kaylarep** [comment]
Is there a way for a hosted group to pause/delete their card? Think this could be useful (or perhaps an option to submit a request to the host to do it.)

**Comment by alanna** [comment]
I think the receipt upload email should only be sent to the responsible admin, not all admins, and there should be a followup reminder if not done in 7 days to all admins. We're having a confusing time because no one knows which notifications are for who and who has what receipt among multiple admins.

**Comment by kaylarep** [comment]
some feedback from a group:
> One thing with the virtual cards that would be helpful would be somewhere KMA could view the transactions/balances to make sure we are on the same page budget wise. I have to email an OCF member to see those numbers.

**Comment by kaylarep** [comment]
Add to the confirmation blurb:

**Comment by kaylarep** [comment]
In the confirmation email, (if I am understanding the flow correctly),

**Comment by kaylarep** [comment]
In the notification email that the assignee receives, can we add a sentence?:
`You can now view your card number & details in your collective's **Settings -> Virtual Cards**'

**Comment by znarf** [comment]
https://github.com/opencollective/opencollective/issues/5078

**Comment by alanna** [comment]
#5196

**Comment by znarf** [comment]
Virtual Card pages in settings should link to each other https://github.com/opencollective/opencollective/issues/5115

**Comment by BenJam** [comment]
more feedback https://github.com/opencollective/opencollective/issues/5295

**Comment by znarf** [comment]
Similar issue there https://github.com/opencollective/opencollective/issues/5410 duplicate?

**Comment by Betree** [comment]
- https://github.com/opencollective/opencollective/issues/5783

**Comment by kaylarep** [comment]
Apologies if this is repetitive, it keeps coming up in various forms, so here's my latest proposal:
- Should be 1 card/assignee/Project or Collective or Host or Event (like a company card)
   - e.g. I have my own card for using to spend from the OCF fiscal host budget. I have a separate, also my own, OC Retreat Event card.
   - It could also be helpful to create another 'Role' in the Team (name tbd) that's specifically just for virtual card assignees e.g. some mutual aid groups have a person on the team who's only job it is is to buy groceries- it would be great for them to have their own card (better accountability than a group card), and be able to upload their own receipts. 
   - e.g. Pain point: We have run into an issue for the Retreat where some team members who are not 'Admins' of OC would like access to a virtual card. We don't have a good way of giving them this access on the platform.

- Only that assignee has access to that card number, only that assignee gets emailed for receipts.
- That assignee then decides what to do with the card (if they want to share it with anyone);
- Also is responsible for understanding the policy, or ensuring anyone they share the card # with understands the policy
- Add a tickbox at virtual card request: `I understand this card may not be used for paying people`. +link to policy post in platform (not in docs)

To help fiscal host admins:
- They will be notified any time a card is charged >$500 to request a manual confirmation the card is being used appropriately

I'm still not convinced this is exactly the best solution, but am trying to consider what would be scalable to 3rd party hosts. Right now we're using a few workarounds, relying on double-checking to avoid duplication, the team has to ask in slack: "Who's missing receipt is this?", doing a lot of work off-platform in Stripe (e.g. adjusting for granularity of monthly/yearly; updating for >$2000 which happens quite frequently).  I wouldn't necessarily feel comfortable releasing this feature as is to 3rd party hosts- unless they were willing to provide direct feedback about how to improve.

**Comment by kaylarep** [comment]
A couple other notes:

Here's my canned reply I send after creating a card:

>The card has been issued to you with a $__/monthly budget. Please be sure to upload receipts within 30 days and ensure all purchases comply with our [Virtual Card Policy](https://docs.opencollective.foundation/how-it-works/policies/virtual-cards-policy).  If at any point you'd like to request an increase, you can reply here/email us.
---
Also, I haven't found a use for 'Naming' a card, now that cards are tied directly to budgets. We used to name cards when they were vendor locked, but since the change to new card provider, I just Name the card the Name of the Collective (it seems redundant.) I'm not sure if I messed things up by proposing https://github.com/opencollective/opencollective/issues/5692 , but that has been a helpful implementation regardless.

**Comment by kaylarep** [comment]
It could be nice to add an extra option on this drop down, where an admin can see all the existing/assigned cards to this Event/Project/Collective: `View Virtual Cards`

![image](https://user-images.githubusercontent.com/64920686/187244572-ac8edb64-2bd1-41f9-b89c-04c404c15429.png)

**Comment by kaylarep** [comment]
from [user feedback](https://opencollective.slack.com/archives/GSNDVCC4F/p1662331815617469): 

> It would be helpful if the email notifications reference the Collective

![image](https://user-images.githubusercontent.com/64920686/189331932-bdc44f7f-3b52-4b07-976b-a4d1ddd43ee1.png)

**Comment by kaylarep** [comment]
Additional feedback: https://opencollective.slack.com/archives/GSNDVCC4F/p1663156125113359?thread_ts=1663148382.026029&cid=GSNDVCC4F

**Comment by kaylarep** [comment]
Another [request ](https://opencollective.freshdesk.com/a/tickets/111844)to track balances (lmk I should refrain from continuing to post these feedbacks): 
![image](https://user-images.githubusercontent.com/64920686/191016168-6fdc218f-e37f-463d-b4ba-27dfee511875.png)

**Comment by znarf** [comment]
> I have never used the 'Assign virtual card' feature since we migrated to Stripe.

A bit of an edge case, but can still be useful, especially for physical cards.

**Comment by znarf** [comment]
> Also, I haven't found a use for 'Naming' a card, now that cards are tied directly to budgets.

It make sense if Collectives are issued multiple cards for multiple purposes, even if it's not the way OCF and OSC generally issue cards. We could make it optional and have use a default.

**Comment by znarf** [comment]
> to edit the card balance limit > $2000 (this is frequent)

I saw recently an authorization to have a monthly limit north of $20,000. I see a huge danger with high limits if the card gets misused or stolen. Is the Collective or the Host ready to absorb the cost of a significant abuse ?

**Comment by kaylarep** [comment]
> > to edit the card balance limit > $2000 (this is frequent)
> 
> I saw recently an authorization to have a monthly limit north of $20,000. I see a huge danger with high limits if the card gets misused or stolen. Is the Collective or the Host ready to absorb the cost of a significant abuse ?

@alanna @natehn thoughts/help on this for OCF?

**Comment by poohlaga** [comment]
I haven't seen any requests from OSC collectives to increase the limit over 2k

**Comment by alanna** [comment]
This particular $20k one is held by an OCF team member (Sade) for a Collective they are involved with. It's for a one-off event and the card will expire after this narrow use. We discussed it and made the call we were OK with this risk as we know the cardholder very well and the card will expire after this specific use.

**Comment by BenJam** [comment]
W3 notes

taking a look at https://github.com/opencollective/opencollective/issues/5991

**Comment by BenJam** [comment]
FYI I just took a stab at prioritising the lists above, predominantly based on @kaylarep's feedback. Thanks @aminakazi for the prod :)

**Comment by natehn** [comment]
Would it be useful to add Granular limits for Virtual Cards https://github.com/opencollective/opencollective/issues/5992 somewhere here? Or is it outside the scope of this project?

**Comment by aminakazi** [comment]
#6120 also brings up the conversation of having assignee(s) to a card. We need to define an assignee's roles, responsibilities, and expectations.

**Comment by Betree** [comment]
As an interesting side-effect of https://github.com/opencollective/opencollective/issues/3749 on this project, virtual card assignees are now displayed in the expenses list (the small avatar).

![image](https://user-images.githubusercontent.com/1556356/206031993-ff75d2b0-057f-4273-92eb-c9e064bfa99e.png)

**Comment by Betree** [comment]
Would be great to have an empty view, this is what collectives that have not created virtual cards yet see when they first go to Admin > Virtual Cards:

![image](https://user-images.githubusercontent.com/1556356/210072204-1360823e-8422-4080-bbc7-ee9a58805cb8.png)

**Comment by kaylarep** [comment]
https://github.com/opencollective/opencollective/issues/5431#issuecomment-1267185864

**Comment by Betree** [comment]
The label in expenses list is not set

![image](https://github.com/opencollective/opencollective/assets/1556356/72fc4e6e-5e05-4968-bb2c-5a71907c410d)

**Comment by znarf** [comment]
- [ ] Support Open Collective Europe case with custom cardholders

**Comment by BenJam** [comment]
notes from kickoff:

- it is possible for a collective to delete their own card information, any checks should also take into account missing receipts for deleted cards. 
- some collectives are using a card ~2-3 times a year or perhaps yearly
- start 'contact assignee with 'mailto:' then consider form modal
- perhaps make an option for 'number of days' next to auto-pause on virtual cards

future needs discussed:

- assign a card to an individual only
- assign a card to a non-administrator
- pause/remove card access when user leaves collective

Priorities

- audit block first
- 'proactive' block second
- 'operational risk' third
 others are equal

Design
Sounds like support needed on 2, 3rd priorities. will schedule to start next week if starting project next week. 

Eng. 
Current focus on workspace and agreements, so may not be scheduled next week.

**Comment by hdiniz** [comment]
This week
- [ ] Page of pending virtual card applications
- [ ] Migrate host virtual cards to [table view](https://www.figma.com/file/KinuveYf4WUlRIXzMTnmIsmO/%5BOC-Design%5D-Lab-%F0%9F%A7%AA?type=design&node-id=21163-118753&mode=design&t=vvUVHv9lsKeC021N-4)

**Comment by hdiniz** [comment]
This week

- [ ] Release of host virtual cards redesign [PR](https://github.com/opencollective/opencollective-frontend/pull/9097)
- [ ] Release page of pending virtual card applications

**Comment by iamronen** [comment]
Summary review meeting regarding virtual card requests. We agreed:
1. That surfacing visual cards requests without being able to act on them defeats the purpose.
2. To add (within the scope of this cycle):
    - A way to create virtual cards from the list of requests. The information from the request will be used to pre-populate the "create virtual card" form with the credit limitation period (daily/weekly/monthly/yearly), the requested limit and the purpose (as the default name of the card). The fiscal host admin can then update the information and create the card
    - A way to reject a virtual card request (and @hdiniz I think we should add a notification to the submitter that the request has been rejected)
    - To enhance the virtual card request form to enable selection of the credit cycle (daily, weekly, monthly, yearly).
 3. We agreed to NOT add in this scope of work the conversational pattern (comment thread with public/private notes and attachments) but to leave this for a future iteration.
 4. That this feature will be opt-in (behind a feature flag manually activated for hosts who wish to use this feature) so that if hosts do not wish to use (for example, if they need the conversation pattern and have it on Freshdesk and do not wish to duplicate information they then have to then also manage on the platform). attn @BenJam & @jpjhill for release notes and documentation.

@hdiniz There is a chance that hosts who do use choose to use this may ask to stop sending requests to Freshdesk (no need for duplication and this further reduces the load on Freshdesk). Can we do anything to prepare to respond to such a request?

**Comment by iamronen** [comment]
@hdiniz I am looking at the new virtual cards tool that has been made into a table. It is showing 6 items per page (I am guessing this is inherited from the previous card-view). In table view this limitation almost feels like something is wrong ("why didn't it load more cards?"). I think the limit can be raised (do we have a pagination standard that we use regularly?) to a larger number to make the screen more effective.

**Comment by hdiniz** [comment]
Yes, this was left over from the previous card view. There isn't a consistent pagination we use, it varies (10, 15, 20) depending on the page. I increased the default pagination of the host virtual card page to 20 as the information is more condensed in the table and with this number all items are visible within a page scroll.

It is possible for user to manipulate the query in the url to return a desired limit, like `https://opencollective.com/dashboard/opensource/host-virtual-cards?limit=5`, this is "feature" available in most paginated views in the platform, but not available to users through an UI component. This is something we can maybe work on as part of the filters redesign.

**Comment by BenJam** [comment]
@hdiniz it looks like the z indexing of the card status is off relative to the context menu:


![Image](https://github.com/opencollective/opencollective/assets/158833/98a8d396-c395-4f5f-bb99-db480b418576)

**Comment by BenJam** [comment]
thanks @hdiniz for taking a look at ☝️
